velox/functions/prestosql/tests/ArrayNGramTest.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cstdint>
+#include <optional>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class ArrayNGramTest : public FunctionBaseTest {};
+
+TEST_F(ArrayNGramTest, intArrays) {
+  using array_type = std::optional<std::vector<std::optional<int64_t>>>;
+  auto test = [this](
+                  const std::vector<int64_t>& inputArray,
+                  int64_t size,
+                  const std::vector<array_type>& expectedOutput) {
+    auto input = makeArrayVector<int64_t>({inputArray});
+    auto result =
+        evaluate(fmt::format("ngrams(c0, {})", size), makeRowVector({input}));
+
+    auto expected = makeNullableNestedArrayVector<int64_t>({{expectedOutput}});
+    assertEqualVectors(expected, result);
+  };
+
+  test({1, 2, 3, 4}, 1, {{{1}}, {{2}}, {{3}}, {{4}}});
+  test({1, 2, 3, 4}, 2, {{{1, 2}}, {{2, 3}}, {{3, 4}}});
+  test({1, 2, 3, 4}, 3, {{{1, 2, 3}}, {{2, 3, 4}}});
+  test({1, 2, 3, 4}, 4, {{{1, 2, 3, 4}}});
+  test({1, 2, 3, 4}, 5, {{{1, 2, 3, 4}}});
+  test({1, 2, 3, 4}, std::numeric_limits<int32_t>::max(), {{{1, 2, 3, 4}}});
+  test(
+      {1, 2, 3, 4},
+      std::numeric_limits<int32_t>::max() + (int64_t)(1000),
+      {{{1, 2, 3, 4}}});
+  test({}, 1, {{{}}});
+  test({}, 10, {{{}}});
+
+  // Test of invalid size
+  auto input = makeArrayVector<int64_t>({{1, 2, 3, 4}});
+  VELOX_ASSERT_THROW(
+      evaluate("ngrams(c0, 0)", makeRowVector({input})), "N must be positive.");
+  VELOX_ASSERT_THROW(
+      evaluate("ngrams(c0, -5)", makeRowVector({input})),
+      "N must be positive.");
+  input = makeArrayVector<int64_t>({{}});
+  VELOX_ASSERT_THROW(
+      evaluate("ngrams(c0, 0)", makeRowVector({input})), "N must be positive.");
+}
+
+TEST_F(ArrayNGramTest, testNull) {
+  using output_type = std::optional<std::vector<std::optional<std::string>>>;
```

Comment:
Nit : Have a templated function for the test lambda in the parent test class to avoid the code repetition.